psiminer-core/src/main/kotlin/labelextractor/MethodCommentLabelExtractor.kt

```diff
@@ -0,0 +1,29 @@
+package labelextractor
+
+import astminer.common.splitToSubtokens
+import com.intellij.psi.PsiComment
+import com.intellij.psi.PsiElement
+import com.intellij.psi.javadoc.PsiDocComment
+import com.intellij.psi.util.PsiTreeUtil
+import psi.language.LanguageHandler
+
+class MethodCommentLabelExtractor : LabelExtractor() {
+    override val granularityLevel = GranularityLevel.Method
+
+    override fun handleTree(root: PsiElement, languageHandler: LanguageHandler): Label? {
+        val methodCommentNodes = PsiTreeUtil.collectElementsOfType(root, PsiComment::class.java)
```

Comment:
Comments in some languages can have their own type hierarchy. Thus, let's use `languageHandler.methodProvider` to handle comment extraction in different languages.

While in Java and Kotlin the comments indeed implement `PsiComment` interface this way or another, in Python it only holds for inline comments. Consider the following snippet of Python code:
```python
def fun():
    """
    comment-1
    """
    ...
    # comment-2
    ...
    """
    comment-3
    """
``` 
Here, `comment-2` is `PsiComment`, `comment-1` is `PsiElement(Py:DOCSTRING)`, and `comment-3` is `PsiElement(Py:TRIPLE_QUOTED_STRING)`. Thus, you won't detect the actual docstring in Python with this approach.